pkg/resources/pods.go

```diff
@@ -99,33 +99,70 @@ func (pa PodAccessor) NotReadyCount() (int, error) {
 	return nr, err
 }
 
-// PodIPsByAge returns the list of running pod (terminating
-// and non-running are excluded) IP addresses, sorted descending by pod age.
-func (pa PodAccessor) PodIPsByAge() ([]string, error) {
+// PodFilter provides a way to filter pods for a revision.
+// Returning true, means that pod should be kept.
+type PodFilter func(p *corev1.Pod) bool
+
+// PodTransformer provides a way to do something with the pod
+// that has been selected by the filters.
+// For example pod transformer may extract a field and store it in
+// internal state.
+type PodTransformer func(p *corev1.Pod)
+
+// ProcessProd filters all the pods using provided pod filters and then if the pod
+// is selected, applies the transformer to it.
+func (pa PodAccessor) ProcessPods(pt PodTransformer, pfs ...PodFilter) error {
 	pods, err := pa.podsLister.List(pa.selector)
 	if err != nil {
-		return nil, err
+		return err
 	}
-	// Keep only running ones.
-	write := 0
 	for _, p := range pods {
-		if p.Status.Phase == corev1.PodRunning && p.DeletionTimestamp == nil {
-			pods[write] = p
-			write++
+		take := true
+		for _, pf := range pfs {
```

Comment:
The inner loop as a helper func might be neater to read (instead of the take bool, you just return)

Reply:
Sounds good.